warehouse/integrations/redshift/redshift.go

```diff
@@ -862,37 +862,33 @@ func (rs *Redshift) loadUserTables(ctx context.Context) map[string]error {
 	}
 	defer rs.dropStagingTables(ctx, []string{stagingTableName})
 
-	primaryKey := "id"
-	query = fmt.Sprintf(`
-		DELETE FROM
-		  %[1]s.%[2]q USING %[1]s.%[3]q _source
-		WHERE
-		  (
-			_source.%[4]s = %[1]s.%[2]s.%[4]s
-		  );
-`,
-		rs.Namespace,
-		warehouseutils.UsersTable,
-		stagingTableName,
-		primaryKey,
-	)
+	if rs.enableDedup() {
```

Comment:
We weren't checking any flags for dedup here, just pinging you guys to make sure it makes sense to do so before I add the tests. @lvrach @achettyiitr